checks/src/main/java/com/kozaxinan/android/checks/NetworkLayerClassSerializedNameDetector.kt

```diff
@@ -22,40 +22,58 @@ internal class NetworkLayerClassSerializedNameDetector : RetrofitReturnTypeDetec
     class NetworkLayerDtoFieldVisitor(private val context: JavaContext) : Visitor(context) {
 
         override fun visitMethod(node: UMethod) {
-            val nonFinalFields = findAllFieldsOf(node)
-                    .filterNot { !it.isStatic && it.getContainingUClass()?.isEnum == true }
-                    .filterNot(::hasSerializedNameAnnotation)
-                    .map { it.name }
+            val nonFinalFields = findAllReturnTypeFieldsOf(node)
+                .filterNot { !it.isStatic && it.getContainingUClass()?.isEnum == true }
+                .filterNot(::hasSerializedNameAnnotation)
+                .map { it.name }
 
             if (nonFinalFields.isNotEmpty()) {
                 context.report(
-                        issue = ISSUE_NETWORK_LAYER_CLASS_SERIALIZED_NAME_RULE,
-                        scopeClass = node,
-                        location = context.getNameLocation(node),
-                        message = "Return type doesn't have @SerializedName annotation for $nonFinalFields fields."
+                    issue = ISSUE_NETWORK_LAYER_CLASS_SERIALIZED_NAME_RULE,
+                    scopeClass = node,
+                    location = context.getNameLocation(node),
+                    message = "Return type doesn't have @SerializedName annotation for $nonFinalFields fields."
                 )
             }
+
+            findAllBodyParameterTypeFieldsOf(node)?.let { (classType, fieldsList) ->
```

Comment:
New functionality in this class, the rest is just formatting

Reply:
This is useful, can you also add a couple of test cases for this new feature?